internal/peering-request-operator/foreign-cluster.go

```diff
@@ -0,0 +1,158 @@
+package peering_request_operator
+
+import (
+	"errors"
+	v1 "github.com/liqoTech/liqo/api/discovery/v1"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog"
+	"k8s.io/utils/pointer"
+	"strings"
+)
+
+func (r *PeeringRequestReconciler) UpdateForeignCluster(pr *v1.PeeringRequest) error {
+	tmp, err := r.crdClient.Resource("foreignclusters").Get(pr.Spec.ClusterID, metav1.GetOptions{})
```

Comment:
Does we require both get operations? I think we have to make a clear design choice (e.g. get by label) since we have control on how the ForeignCluster are created or am I missing something?

Reply:
In 90% of times (every time that is not manually create) we can use Get function that is faster, if get fails we can try to get it by label

If it seems too much confusing I can use label only

Reply:
Which are the case which are not accessible using a get by name?

Reply:
Only if someone creates it manually